File: Chapter_01_Introduction/Getting_Started_The_Easy_Problems/Easy/11559-Event_Planning/main.cpp
/*
 * Problem 11559 - Event Planning
 */
#include <iostream>
#include <sstream>

int main() {
    std::ios::sync_with_stdio();
    std::cin.tie(0);

    std::ostringstream output;
    int participants;
    int budget;
    int hotels;
    int weeks;
    while (std::cin >> participants >> budget >> hotels >> weeks) {
        int cheapest = budget + 1;
        for (int i = 0; i < hotels; ++i) {
            int pricePerPerson;
            std::cin >> pricePerPerson;
            int cost {pricePerPerson * participants};
            bool inBudget = cost <= budget;
            bool vacancy {};
            for (int j = 0; j < weeks; ++j) {
                int rooms;
                std::cin >> rooms;
                if (rooms >= participants) {
                    vacancy = true;
                }
            }
            cheapest = (inBudget && vacancy && cost < cheapest) ? cost : cheapest;
        }
        if (cheapest > budget) {
            output << "stay home\n";
        } else {
            output << cheapest << '\n';
        }
    }
    std::cout << output.str();
}

File: Chapter_01_Introduction/Getting_Started_The_Easy_Problems/Easy/11559-Event_Planning/Main.java
/*
 * Problem 11559 - Event Planning
 * Java version 6/12/2016
 */

import java.io.PrintStream;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        StringBuffer results = new StringBuffer();

        while (input.hasNext()) {
            int numberOfParticipants = input.nextInt();
            int budget = input.nextInt();
            int numberOfHotels = input.nextInt();
            int numberOfWeeks = input.nextInt();

            int cost = budget + 1;
            for (int i = 0; i < numberOfHotels; ++i) {
                int totalCost = numberOfParticipants * input.nextInt();
                for (int j = 0; j < numberOfWeeks; ++j) {
                    int beds = input.nextInt();
                    if (beds >= numberOfParticipants) {
                        if (totalCost < budget && totalCost < cost) {
                            cost = totalCost;
                        }
                    }
                }
            }
            if (cost > budget) {
                results.append("stay home\n");
            } else {
                results.append(cost + "\n");
            }
        }
        System.out.print(results.toString());
    }
}
